### ISSUE TITLE
Errors on latest version when using esp8266

### ISSUE THREAD

**Issue by ErrorErrorError** [bblanchon/ArduinoStreamUtils#2]
Hello, 

I recently learned about this library after using ArduinoJSON and when I use this library and compile, I get this error 

> In file included from /Users/erikb/.platformio/lib/StreamUtils_ID6347/src/StreamUtils/Clients/LoggingClient.hpp:7:0,
from /Users/erikb/.platformio/lib/StreamUtils_ID6347/src/StreamUtils.h:5,
from src/main.cpp:6:
/Users/erikb/.platformio/lib/StreamUtils_ID6347/src/StreamUtils/Clients/../Policies/ConnectForwardingPolicy.hpp: In member function 'bool StreamUtils::ConnectForwardingPolicy::stop(Client&, unsigned int)':
/Users/erikb/.platformio/lib/StreamUtils_ID6347/src/StreamUtils/Clients/../Policies/ConnectForwardingPolicy.hpp:28:31: error: no matching function for call to 'Client::stop(unsigned int&)'
return target.stop(timeout);
^
/Users/erikb/.platformio/lib/StreamUtils_ID6347/src/StreamUtils/Clients/../Policies/ConnectForwardingPolicy.hpp:28:31: note: candidate is:
In file included from /Users/erikb/.platformio/packages/framework-arduinoespressif8266/libraries/ESP8266WiFi/src/WiFiClient.h:27:0,
from /Users/erikb/.platformio/packages/framework-arduinoespressif8266/libraries/ESP8266WiFi/src/ESP8266WiFi.h:39,
from src/main.cpp:4:
/Users/erikb/.platformio/packages/framework-arduinoespressif8266/cores/esp8266/Client.h:38:22: note: virtual void Client::stop()
virtual void stop() = 0;
^
/Users/erikb/.platformio/packages/framework-arduinoespressif8266/cores/esp8266/Client.h:38:22: note:   candidate expects 0 arguments, 1 provided
In file included from /Users/erikb/.platformio/lib/StreamUtils_ID6347/src/StreamUtils/Clients/LoggingClient.hpp:9:0,
from /Users/erikb/.platformio/lib/StreamUtils_ID6347/src/StreamUtils.h:5,
from src/main.cpp:6:
/Users/erikb/.platformio/lib/StreamUtils_ID6347/src/StreamUtils/Clients/../Policies/WriteLoggingPolicy.hpp: In member function 'bool StreamUtils::WriteLoggingPolicy::flush(Client&, unsigned int)':
/Users/erikb/.platformio/lib/StreamUtils_ID6347/src/StreamUtils/Clients/../Policies/WriteLoggingPolicy.hpp:34:32: error: no matching function for call to 'Client::flush(unsigned int&)'
return target.flush(timeout);
^
/Users/erikb/.platformio/lib/StreamUtils_ID6347/src/StreamUtils/Clients/../Policies/WriteLoggingPolicy.hpp:34:32: note: candidate is:
In file included from /Users/erikb/.platformio/packages/framework-arduinoespressif8266/libraries/ESP8266WiFi/src/WiFiClient.h:27:0,
from /Users/erikb/.platformio/packages/framework-arduinoespressif8266/libraries/ESP8266WiFi/src/ESP8266WiFi.h:39,
from src/main.cpp:4:
/Users/erikb/.platformio/packages/framework-arduinoespressif8266/cores/esp8266/Client.h:37:22: note: virtual void Client::flush()
virtual void flush() = 0;
^
/Users/erikb/.platformio/packages/framework-arduinoespressif8266/cores/esp8266/Client.h:37:22: note:   candidate expects 0 arguments, 1 provided
In file included from /Users/erikb/.platformio/lib/StreamUtils_ID6347/src/StreamUtils/Clients/LoggingClient.hpp:10:0,
from /Users/erikb/.platformio/lib/StreamUtils_ID6347/src/StreamUtils.h:5,
from src/main.cpp:6:
/Users/erikb/.platformio/lib/StreamUtils_ID6347/src/StreamUtils/Clients/ClientProxy.hpp: In instantiation of 'class StreamUtils::ClientProxy<StreamUtils::ReadLoggingPolicy, StreamUtils::WriteLoggingPolicy, StreamUtils::ConnectForwardingPolicy>':
/Users/erikb/.platformio/lib/StreamUtils_ID6347/src/StreamUtils/Clients/LoggingClient.hpp:14:24:   required from here
/Users/erikb/.platformio/lib/StreamUtils_ID6347/src/StreamUtils/Clients/ClientProxy.hpp:69:7: error: 'int StreamUtils::ClientProxy<ReadPolicy, WritePolicy, ConnectPolicy>::connect(const IPAddress&, uint16_t) [with ReadPolicy = StreamUtils::ReadLoggingPolicy; WritePolicy = StreamUtils::WriteLoggingPolicy; ConnectPolicy = StreamUtils::ConnectForwardingPolicy; uint16_t = short unsigned int]' marked override, but does not override
int connect(const IPAddress &ip, uint16_t port) override {
^
/Users/erikb/.platformio/lib/StreamUtils_ID6347/src/StreamUtils/Clients/ClientProxy.hpp:85:8: error: 'bool StreamUtils::ClientProxy<ReadPolicy, WritePolicy, ConnectPolicy>::stop(unsigned int) [with ReadPolicy = StreamUtils::ReadLoggingPolicy; WritePolicy = StreamUtils::WriteLoggingPolicy; ConnectPolicy = StreamUtils::ConnectForwardingPolicy]' marked override, but does not override
bool stop(unsigned timeout = 0) override {
^
/Users/erikb/.platformio/lib/StreamUtils_ID6347/src/StreamUtils/Clients/ClientProxy.hpp:105:8: error: 'bool StreamUtils::ClientProxy<ReadPolicy, WritePolicy, ConnectPolicy>::flush(unsigned int) [with ReadPolicy = StreamUtils::ReadLoggingPolicy; WritePolicy = StreamUtils::WriteLoggingPolicy; ConnectPolicy = StreamUtils::ConnectForwardingPolicy]' marked override, but does not override
bool flush(unsigned timeout = 0) override {
^
In file included from /Users/erikb/.platformio/lib/StreamUtils_ID6347/src/StreamUtils/Clients/ReadBufferingClient.hpp:9:0,
from /Users/erikb/.platformio/lib/StreamUtils_ID6347/src/StreamUtils.h:6,
from src/main.cpp:6:
/Users/erikb/.platformio/lib/StreamUtils_ID6347/src/StreamUtils/Clients/../Policies/WriteForwardingPolicy.hpp: In member function 'bool StreamUtils::WriteForwardingPolicy::flush(Client&, unsigned int)':
/Users/erikb/.platformio/lib/StreamUtils_ID6347/src/StreamUtils/Clients/../Policies/WriteForwardingPolicy.hpp:26:32: error: no matching function for call to 'Client::flush(unsigned int&)'
return client.flush(timeout);
^
/Users/erikb/.platformio/lib/StreamUtils_ID6347/src/StreamUtils/Clients/../Policies/WriteForwardingPolicy.hpp:26:32: note: candidate is:
In file included from /Users/erikb/.platformio/packages/framework-arduinoespressif8266/libraries/ESP8266WiFi/src/WiFiClient.h:27:0,
from /Users/erikb/.platformio/packages/framework-arduinoespressif8266/libraries/ESP8266WiFi/src/ESP8266WiFi.h:39,
from src/main.cpp:4:
/Users/erikb/.platformio/packages/framework-arduinoespressif8266/cores/esp8266/Client.h:37:22: note: virtual void Client::flush()
virtual void flush() = 0;
^
/Users/erikb/.platformio/packages/framework-arduinoespressif8266/cores/esp8266/Client.h:37:22: note:   candidate expects 0 arguments, 1 provided
In file included from /Users/erikb/.platformio/lib/StreamUtils_ID6347/src/StreamUtils/Clients/LoggingClient.hpp:10:0,
from /Users/erikb/.platformio/lib/StreamUtils_ID6347/src/StreamUtils.h:5,
from src/main.cpp:6:
/Users/erikb/.platformio/lib/StreamUtils_ID6347/src/StreamUtils/Clients/ClientProxy.hpp: In instantiation of 'class StreamUtils::ClientProxy<StreamUtils::ReadLoggingPolicy, StreamUtils::WriteForwardingPolicy, StreamUtils::ConnectForwardingPolicy>':
/Users/erikb/.platformio/lib/StreamUtils_ID6347/src/StreamUtils/Clients/ReadLoggingClient.hpp:14:28:   required from here
/Users/erikb/.platformio/lib/StreamUtils_ID6347/src/StreamUtils/Clients/ClientProxy.hpp:69:7: error: 'int StreamUtils::ClientProxy<ReadPolicy, WritePolicy, ConnectPolicy>::connect(const IPAddress&, uint16_t) [with ReadPolicy = StreamUtils::ReadLoggingPolicy; WritePolicy = StreamUtils::WriteForwardingPolicy; ConnectPolicy = StreamUtils::ConnectForwardingPolicy; uint16_t = short unsigned int]' marked override, but does not override
int connect(const IPAddress &ip, uint16_t port) override {
^
/Users/erikb/.platformio/lib/StreamUtils_ID6347/src/StreamUtils/Clients/ClientProxy.hpp:85:8: error: 'bool StreamUtils::ClientProxy<ReadPolicy, WritePolicy, ConnectPolicy>::stop(unsigned int) [with ReadPolicy = StreamUtils::ReadLoggingPolicy; WritePolicy = StreamUtils::WriteForwardingPolicy; ConnectPolicy = StreamUtils::ConnectForwardingPolicy]' marked override, but does not override
bool stop(unsigned timeout = 0) override {
^
/Users/erikb/.platformio/lib/StreamUtils_ID6347/src/StreamUtils/Clients/ClientProxy.hpp:105:8: error: 'bool StreamUtils::ClientProxy<ReadPolicy, WritePolicy, ConnectPolicy>::flush(unsigned int) [with ReadPolicy = StreamUtils::ReadLoggingPolicy; WritePolicy = StreamUtils::WriteForwardingPolicy; ConnectPolicy = StreamUtils::ConnectForwardingPolicy]' marked override, but does not override
bool flush(unsigned timeout = 0) override {
^
In file included from /Users/erikb/.platformio/lib/StreamUtils_ID6347/src/StreamUtils/Clients/WriteBufferingClient.hpp:9:0,
from /Users/erikb/.platformio/lib/StreamUtils_ID6347/src/StreamUtils.h:8,
from src/main.cpp:6:
/Users/erikb/.platformio/lib/StreamUtils_ID6347/src/StreamUtils/Clients/../Policies/WriteBufferingPolicy.hpp: In member function 'bool StreamUtils::WriteBufferingPolicy<TAllocator>::flush(Client&, unsigned int)':
/Users/erikb/.platformio/lib/StreamUtils_ID6347/src/StreamUtils/Clients/../Policies/WriteBufferingPolicy.hpp:73:32: error: no matching function for call to 'Client::flush(unsigned int&)'
return target.flush(timeout);
^
/Users/erikb/.platformio/lib/StreamUtils_ID6347/src/StreamUtils/Clients/../Policies/WriteBufferingPolicy.hpp:73:32: note: candidate is:
In file included from /Users/erikb/.platformio/packages/framework-arduinoespressif8266/libraries/ESP8266WiFi/src/WiFiClient.h:27:0,
from /Users/erikb/.platformio/packages/framework-arduinoespressif8266/libraries/ESP8266WiFi/src/ESP8266WiFi.h:39,
from src/main.cpp:4:
/Users/erikb/.platformio/packages/framework-arduinoespressif8266/cores/esp8266/Client.h:37:22: note: virtual void Client::flush()
virtual void flush() = 0;
^
/Users/erikb/.platformio/packages/framework-arduinoespressif8266/cores/esp8266/Client.h:37:22: note:   candidate expects 0 arguments, 1 provided
In file included from /Users/erikb/.platformio/lib/StreamUtils_ID6347/src/StreamUtils/Clients/LoggingClient.hpp:10:0,
from /Users/erikb/.platformio/lib/StreamUtils_ID6347/src/StreamUtils.h:5,
from src/main.cpp:6:
/Users/erikb/.platformio/lib/StreamUtils_ID6347/src/StreamUtils/Clients/ClientProxy.hpp: In instantiation of 'class StreamUtils::ClientProxy<StreamUtils::ReadForwardingPolicy, StreamUtils::WriteLoggingPolicy, StreamUtils::ConnectForwardingPolicy>':
/Users/erikb/.platformio/lib/StreamUtils_ID6347/src/StreamUtils/Clients/WriteLoggingClient.hpp:15:7:   required from here
/Users/erikb/.platformio/lib/StreamUtils_ID6347/src/StreamUtils/Clients/ClientProxy.hpp:69:7: error: 'int StreamUtils::ClientProxy<ReadPolicy, WritePolicy, ConnectPolicy>::connect(const IPAddress&, uint16_t) [with ReadPolicy = StreamUtils::ReadForwardingPolicy; WritePolicy = StreamUtils::WriteLoggingPolicy; ConnectPolicy = StreamUtils::ConnectForwardingPolicy; uint16_t = short unsigned int]' marked override, but does not override
int connect(const IPAddress &ip, uint16_t port) override {
^
/Users/erikb/.platformio/lib/StreamUtils_ID6347/src/StreamUtils/Clients/ClientProxy.hpp:85:8: error: 'bool StreamUtils::ClientProxy<ReadPolicy, WritePolicy, ConnectPolicy>::stop(unsigned int) [with ReadPolicy = StreamUtils::ReadForwardingPolicy; WritePolicy = StreamUtils::WriteLoggingPolicy; ConnectPolicy = StreamUtils::ConnectForwardingPolicy]' marked override, but does not override
bool stop(unsigned timeout = 0) override {
^
/Users/erikb/.platformio/lib/StreamUtils_ID6347/src/StreamUtils/Clients/ClientProxy.hpp:105:8: error: 'bool StreamUtils::ClientProxy<ReadPolicy, WritePolicy, ConnectPolicy>::flush(unsigned int) [with ReadPolicy = StreamUtils::ReadForwardingPolicy; WritePolicy = StreamUtils::WriteLoggingPolicy; ConnectPolicy = StreamUtils::ConnectForwardingPolicy]' marked override, but does not override
bool flush(unsigned timeout = 0) override {
^
*** [.pioenvs/d1_mini/src/main.cpp.o] Error 1

I am using a Wemos D1 R2 and the ide I am using is Visual Studio with PlatformIO but this error persists in Arduino IDE and CLion. Is there any way to fix this? Thank you. 

Code:
`
#include <ESP8266WiFi.h>
#include <ArduinoJson.h>
#include <StreamUtils.h>

const char *ssid = "-----------";
const char *password = "-----------";
const uint8_t port = 80;

WiFiServer wifiServer(port);

const size_t capacity = JSON_OBJECT_SIZE(9) + 110;
DynamicJsonDocument doc(capacity);

void waves(DynamicJsonDocument doc);
void music(DynamicJsonDocument doc);
void solid(DynamicJsonDocument doc);

String t;

uint8_t brightness;

void setup() {

    Serial.begin(115200);

    delay(1000);

    WiFi.begin(ssid, password);

    while (WiFi.status() != WL_CONNECTED) {
        delay(1000);
        Serial.println("Connecting..");
    }

    Serial.print("Connected to WiFi. IP:");
    Serial.println(WiFi.localIP());
    Serial.print("Port:");
    Serial.println(port);

    wifiServer.begin();
}

void loop() {

    WiFiClient client = wifiServer.available();
    if (client) {

        while (client.connected() > 0) {
           // char l = client.read();
           // Serial.print(l);
            ReadBufferingStream bufferingClient(client, capacity);
            deserializeJson(doc, bufferingClient);
            
            t = doc["mode"].as<String>();
            if(t.equals("DeviceMusic")) {
                music(doc);
            } else if(t.equals("DeviceSolid")){
                solid(doc);
            } else if(t.equals("DeviceWaves")){
                waves(doc);
            }
        }
        Serial.println(" ");
        client.flush();
        client.stop();
        delay(10);
    }
}

void waves(DynamicJsonDocument document) {

}

void solid(DynamicJsonDocument doc){
    brightness = doc["brightness"].as<int8_t>();

}

void music(DynamicJsonDocument doc){

}
`

**Comment by bblanchon** [comment]
Hi @ErrorErrorError,

Thank you for reporting this.
It seems that the team reverted the quirks they inserted in the ESP8266 core:
https://github.com/esp8266/Arduino/commit/cdb549572d25cb1abf0658ff285009d591db1f3a#diff-6b1b07c053c3e5f501aa2b480d825226
That's cool, I can remove my workarounds then.

Meanwhile, you can disable the workarounds with these settings:

```c++
#define STREAMUTILS_CLIENT_CONNECT_TAKE_CONST_REF 0
#define STREAMUTILS_CLIENT_FLUSH_TAKES_TIMEOUT 0
#define STREAMUTILS_CLIENT_STOP_TAKES_TIMEOUT 0
#include <StreamUtils.h>
```

Best Regards,
Benoit

**Comment by ErrorErrorError** [comment]
@bblanchon Thank you very much!

**Comment by AngeloEGMdeSousa** [comment]
Hello @bblanchon,
I had this similar error:

In file included from /mnt/create-efs/webide/1c/e2/1ce2f752a26da7808c22fbd4e6f3fd50:bluemot1on/libraries_v2/StreamUtils/src/StreamUtils/Clients/HammingClient.hpp:11:0,

from /mnt/create-efs/webide/1c/e2/1ce2f752a26da7808c22fbd4e6f3fd50:bluemot1on/libraries_v2/StreamUtils/src/StreamUtils.hpp:5,

from /mnt/create-efs/webide/1c/e2/1ce2f752a26da7808c22fbd4e6f3fd50:bluemot1on/libraries_v2/StreamUtils/src/StreamUtils.h:5,

from /tmp/652260870/Projeto_Rega_IoT_aug31a/Projeto_Rega_IoT_aug31a.ino:3:

/mnt/create-efs/webide/1c/e2/1ce2f752a26da7808c22fbd4e6f3fd50:bluemot1on/libraries_v2/StreamUtils/src/StreamUtils/Clients/ClientProxy.hpp: In instantiation of 'class StreamUtils::ClientProxy<StreamUtils::ReadLoggingPolicy, StreamUtils::WriteLoggingPolicy, StreamUtils::ConnectForwardingPolicy>':

/mnt/create-efs/webide/1c/e2/1ce2f752a26da7808c22fbd4e6f3fd50:bluemot1on/libraries_v2/StreamUtils/src/StreamUtils/Clients/LoggingClient.hpp:14:24: required from here

/mnt/create-efs/webide/1c/e2/1ce2f752a26da7808c22fbd4e6f3fd50:bluemot1on/libraries_v2/StreamUtils/src/StreamUtils/Clients/ClientProxy.hpp:76:7: error: 'int StreamUtils::ClientProxy<ReadPolicy, WritePolicy, ConnectPolicy>::connect(IPAddress, uint16_t) [with ReadPolicy = StreamUtils::ReadLoggingPolicy; WritePolicy = StreamUtils::WriteLoggingPolicy; ConnectPolicy = StreamUtils::ConnectForwardingPolicy; uint16_t = short unsigned int]' marked override, but does not override

int connect(IPAddress ip, uint16_t port) override {

^

/mnt/create-efs/webide/1c/e2/1ce2f752a26da7808c22fbd4e6f3fd50:bluemot1on/libraries_v2/StreamUtils/src/StreamUtils/Clients/ClientProxy.hpp:88:8: error: 'void StreamUtils::ClientProxy<ReadPolicy, WritePolicy, ConnectPolicy>::stop() [with ReadPolicy = StreamUtils::ReadLoggingPolicy; WritePolicy = StreamUtils::WriteLoggingPolicy; ConnectPolicy = StreamUtils::ConnectForwardingPolicy]' marked override, but does not override

void stop() override {

^

/mnt/create-efs/webide/1c/e2/1ce2f752a26da7808c22fbd4e6f3fd50:bluemot1on/libraries_v2/StreamUtils/src/StreamUtils/Clients/ClientProxy.hpp: In instantiation of 'class StreamUtils::ClientProxy<StreamUtils::ReadLoggingPolicy, StreamUtils::WriteForwardingPolicy, StreamUtils::ConnectForwardingPolicy>':

/mnt/create-efs/webide/1c/e2/1ce2f752a26da7808c22fbd4e6f3fd50:bluemot1on/libraries_v2/StreamUtils/src/StreamUtils/Clients/ReadLoggingClient.hpp:14:28: required from here

/mnt/create-efs/webide/1c/e2/1ce2f752a26da7808c22fbd4e6f3fd50:bluemot1on/libraries_v2/StreamUtils/src/StreamUtils/Clients/ClientProxy.hpp:76:7: error: 'int StreamUtils::ClientProxy<ReadPolicy, WritePolicy, ConnectPolicy>::connect(IPAddress, uint16_t) [with ReadPolicy = StreamUtils::ReadLoggingPolicy; WritePolicy = StreamUtils::WriteForwardingPolicy; ConnectPolicy = StreamUtils::ConnectForwardingPolicy; uint16_t = short unsigned int]' marked override, but does not override

int connect(IPAddress ip, uint16_t port) override {

^

/mnt/create-efs/webide/1c/e2/1ce2f752a26da7808c22fbd4e6f3fd50:bluemot1on/libraries_v2/StreamUtils/src/StreamUtils/Clients/ClientProxy.hpp:88:8: error: 'void StreamUtils::ClientProxy<ReadPolicy, WritePolicy, ConnectPolicy>::stop() [with ReadPolicy = StreamUtils::ReadLoggingPolicy; WritePolicy = StreamUtils::WriteForwardingPolicy; ConnectPolicy = StreamUtils::ConnectForwardingPolicy]' marked override, but does not override

void stop() override {

^

/mnt/create-efs/webide/1c/e2/1ce2f752a26da7808c22fbd4e6f3fd50:bluemot1on/libraries_v2/StreamUtils/src/StreamUtils/Clients/ClientProxy.hpp: In instantiation of 'class StreamUtils::ClientProxy<StreamUtils::ReadForwardingPolicy, StreamUtils::WriteLoggingPolicy, StreamUtils::ConnectForwardingPolicy>':

/mnt/create-efs/webide/1c/e2/1ce2f752a26da7808c22fbd4e6f3fd50:bluemot1on/libraries_v2/StreamUtils/src/StreamUtils/Clients/WriteLoggingClient.hpp:15:7: required from here

/mnt/create-efs/webide/1c/e2/1ce2f752a26da7808c22fbd4e6f3fd50:bluemot1on/libraries_v2/StreamUtils/src/StreamUtils/Clients/ClientProxy.hpp:76:7: error: 'int StreamUtils::ClientProxy<ReadPolicy, WritePolicy, ConnectPolicy>::connect(IPAddress, uint16_t) [with ReadPolicy = StreamUtils::ReadForwardingPolicy; WritePolicy = StreamUtils::WriteLoggingPolicy; ConnectPolicy = StreamUtils::ConnectForwardingPolicy; uint16_t = short unsigned int]' marked override, but does not override

int connect(IPAddress ip, uint16_t port) override {

^

/mnt/create-efs/webide/1c/e2/1ce2f752a26da7808c22fbd4e6f3fd50:bluemot1on/libraries_v2/StreamUtils/src/StreamUtils/Clients/ClientProxy.hpp:88:8: error: 'void StreamUtils::ClientProxy<ReadPolicy, WritePolicy, ConnectPolicy>::stop() [with ReadPolicy = StreamUtils::ReadForwardingPolicy; WritePolicy = StreamUtils::WriteLoggingPolicy; ConnectPolicy = StreamUtils::ConnectForwardingPolicy]' marked override, but does not override

void stop() override {

^

Error during build: exit status 1


I'm using NodeMCU with this code:

// StreamUtils - Version: 1.6.3
#include <StreamUtils.h>
#include <StreamUtils.hpp>
#define STREAMUTILS_CLIENT_CONNECT_TAKE_CONST_REF 0
#define STREAMUTILS_CLIENT_FLUSH_TAKES_TIMEOUT 0
#define STREAMUTILS_CLIENT_STOP_TAKES_TIMEOUT 0
#define ARDUINOJSON_ENABLE_PROGMEM 0
#include "thingProperties.h"
#include <SoftwareSerial.h>
#include <ArduinoJson.h>
SoftwareSerial mySerial (D6, D5);

int PA;
int RD;
int ACK, ACK1;

const int capacity1 = 32;
const int capacity2 = 192;

void setup() {
  // Initialize serial and wait for port to open:
  Serial.begin(9600);
  mySerial.begin(600);
  // This delay gives the chance to wait for a Serial Monitor without blocking if none is found
  delay(1500); 

  // Defined in thingProperties.h
  initProperties();

  // Connect to Arduino IoT Cloud
  ArduinoCloud.begin(ArduinoIoTPreferredConnection);
  
  /*
     The following function allows you to obtain more information
     related to the state of network and IoT Cloud connection and errors
     the higher number the more granular information you’ll get.
     The default is 0 (only errors).
     Maximum is 4
 */
  setDebugMessageLevel(2);
  ArduinoCloud.printDebugInfo();
  
      // Wait for connection with Arduino IoT Cloud
    while (!ArduinoCloud.connected()) {
    delay(3000);
    Serial.println("NOT connected to the cloud yet...");
    ArduinoCloud.update();
  }
}

void MasterTxRx(){
  static short int state;

  switch (state){

    case 0: {//Envia tipo de dados
    tipo=0;
        StaticJsonDocument<capacity1> tipodoc;
        
        Serial.print("....................................................................");
        Serial.print("\n");
        Serial.print("state = 0");
        Serial.print("--------------------------------------------------------------------");
        Serial.print("\n");
        
        tipodoc["tipo"] = tipo;
        serializeJson(tipodoc, mySerial);
        serializeJsonPretty(tipodoc, Serial);

        Serial.print("\n");
        Serial.print("Enviou o tipo de documento");
        Serial.print("\n");
      
      if(tipo==0){state=1;}
      if(tipo==1){state=3;}
    }
    break;

    case 1:{
      if(mySerial.available() != 0) {state=2;}
      //Serial.println("state = 1");
    }
    break;

    case 2:{ 
      StaticJsonDocument<capacity2> infosondas;
        
      Serial.print("\n");
      Serial.print("....................................................................");
      Serial.print("\n");
      Serial.print("state = 2");
      Serial.print("--------------------------------------------------------------------");
      Serial.print("\n");
      
      //ReadBufferingStream bufferedStream(mySerial, 64);
      //mySerial.setTimeout(10000);
      DeserializationError error = deserializeJson(infosondas, mySerial); 
       if (error) {
       Serial.print("deserializeJson() failed with code ");
       Serial.println(error.c_str());
       return;
       }
       
          sonda1 = infosondas["sonda1"];
          sonda2 = infosondas["sonda2"];
          sonda3 = infosondas["sonda3"];
          sonda4 = infosondas["sonda4"];
          sonda5 = infosondas["sonda5"];
          sonda6 = infosondas["sonda6"];

          Serial.println("Sonda1=");
          Serial.println(sonda1);

        state=0;
    }
    break;
    
    case 3:{
      if(mySerial.available() != 0) {state=4;}
      Serial.println("waiting for serial...");
    }
    break;
    
    case 4:{ //Recebe ACK do recebimento do tipo
      StaticJsonDocument<capacity1> tipoisok;

      Serial.print("....................................................................");
      Serial.print("state = 4");
      Serial.print("\n"); 
      Serial.print("--------------------------------------------------------------------");
      Serial.print("\n");

      DeserializationError error = deserializeJson(tipoisok, mySerial); 
       if (error) {
       Serial.print("deserializeJson() failed with code ");
       Serial.println(error.c_str());
       return;
       }

       ACK = tipoisok["ACK"];

       Serial.println("ACK=");
       Serial.println(ACK);

      if(ACK==1){state=5;}
    }
    break;
    
    case 5:{ //Envia Comandos
      StaticJsonDocument<capacity2> comandos;

      Serial.print("....................................................................");
      Serial.print("state = 5");
      Serial.print("\n"); 
      Serial.print("--------------------------------------------------------------------");
      Serial.print("\n");

        comandos["horaderega"] = horaderega;
        comandos["minutoderega"] = minutoderega;
        comandos["HR"] = HR;
        comandos["MR"] = MR;
        comandos["limiar_umidade"] = limiar_umidade;
        
        serializeJson(comandos, mySerial);
        serializeJsonPretty(comandos, Serial);
      
      state=6;
    }
    break;

    case 6:{
      if(mySerial.available() != 0) {state=7;}
      Serial.println("waiting for serial...");
    }
    break;

    case 7:{
      StaticJsonDocument<capacity1> comandosok;

      Serial.print("....................................................................");
      Serial.print("state = 7");
      Serial.print("\n"); 
      Serial.print("--------------------------------------------------------------------");
      Serial.print("\n");

      DeserializationError error = deserializeJson(comandosok, mySerial); 
       if (error) {
       Serial.print("deserializeJson() failed with code ");
       Serial.println(error.c_str());
       return;
       }

       ACK1 = comandosok["ACK1"];

      if(ACK1==1){state=0;}
    }
    break;
  }
}

void loop() {
  ArduinoCloud.update();
  
  MasterTxRx();
}

What could be the problem?

Thank you.

**Comment by bblanchon** [comment]
@AngeloEGMdeSousa, I just tested with [ESP8266 core](https://github.com/esp8266/Arduino) version 3.0.2, and it worked fine.

Are you using another core?
If so, [**open a new issue**](https://github.com/bblanchon/ArduinoStreamUtils/issues/new/choose) and include the core's name and version.
